plugins-compat-tester/src/main/java/org/jenkins/tools/test/util/ExecutedTestNamesSolver.java

```diff
@@ -108,19 +109,38 @@ public ExecutedTestNamesDetails solve(Set<String> executedTests, File baseDirect
         return testNames;
     }
 
-    private List<String> getSurefireReportsDirectoryPaths(File baseDirectory) throws ExecutedTestNamesSolverException {
+    private List<String> getReportsDirectoryPaths(Set<String> types, File baseDirectory) throws ExecutedTestNamesSolverException {
         List<String> paths = new LinkedList<>();
-        try (Stream<Path> walk = Files.walk(Paths.get(baseDirectory.getAbsolutePath()))) {
-            List<Path> result = walk.filter(Files::isDirectory)
-                    .filter(file -> file.getFileName().toString().endsWith("surefire-reports"))
-                    .collect(Collectors.toList());
-            for (Path path : result) {
-                paths.add(path.toString());
-            }
-        } catch (IOException e) {
-            throw new ExecutedTestNamesSolverException(e);
-        } 
+        if (types == null) {
+            return paths;
+        }
+        for(String type: types) {
+            try (Stream<Path> walk = Files.walk(Paths.get(baseDirectory.getAbsolutePath()))) {
+                List<Path> result = walk.filter(Files::isDirectory)
+                        .filter(file -> file.getFileName().toString().endsWith(String.format("%s-reports", type)))
+                        .collect(Collectors.toList());
+                for (Path path : result) {
+                    paths.add(path.toString());
+                }
+            } catch (IOException e) {
+                throw new ExecutedTestNamesSolverException(e);
+            } 
+        }
         return paths;
     }
     
+    private boolean containsFailure(NodeList nodeList) {
+        if (nodeList.getLength() == 0) {
+            return false;
+        }
+        
+        for (int j = 0; j < nodeList.getLength(); j++) {
+            if (nodeList.item(j).getNodeName().equals("skipped")) {
+                return false;
```

Comment:
So if there is one skipped test, this is saying there are no test failures?

Reply:
Only the specific test that is skipped will not be considered as a failure. Before this PR it was considered as a failure.